src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -440,15 +440,15 @@ typename MatType::elem_type RNN<
     SetCurrentStep(t - extraSteps + 1);
 
     // Wrap a matrix around our data to avoid a copy.
-    MakeAlias(stepData, predictors.slice(t).colptr(begin), predictors.n_rows,
-        batchSize);
-    MakeAlias(outputData, outputs.slice(t).memptr(), outputs.n_rows,
-        outputs.n_cols);
+    MakeAlias(stepData, predictors.slice(t), predictors.n_rows, batchSize,
+        begin * predictors.slice(t).n_rows);
+    MakeAlias(outputData, outputs.slice(t), outputs.n_rows, outputs.n_cols);
     network.network.Forward(stepData, outputData);
 
     const size_t responseStep = (single) ? 0 : t;
-    MakeAlias(responseData, responses.slice(responseStep).colptr(begin),
-        responses.n_rows, batchSize);
+    MakeAlias(responseData, responses.slice(responseStep),
+        responses.n_rows, batchSize,
+        begin * responses.slice(responseStep).n_rows);
```

Comment:
You could just use `responses.n_rows` here (and elsewhere in this file), since it's the same whether or not you take the slice.